pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -1003,13 +1027,45 @@ func (r *Reconciler) runReconcileSeedFlow(
 				Name: "Deploying ETCD Druid",
 				Fn:   etcdDruid.Deploy,
 			})
+			_ = g.Add(flow.Task{
+				Name: "Deploying Fluent Operator Resources and wait until become ready",
+				Fn:   component.OpWait(fluentOperatorResources).Deploy,
+			})
+			_ = g.Add(flow.Task{
+				Name: "Deploying Fluent Operator",
+				Fn:   component.OpWait(fluentOperator).Deploy,
+			})
 		)
 	}
 
 	if err := g.Compile().Run(ctx, flow.Opts{Log: log}); err != nil {
 		return flow.Errors(err)
 	}
 
+	if loggingEnabled {
+		// Waiting for fluent-bit service to be created
+		fluentBitService := &corev1.Service{}
+		if err := retry.UntilTimeout(ctx, 5*time.Second, time.Minute, func(ctx context.Context) (done bool, err error) {
+			if err = seedClient.Get(ctx, kubernetesutils.Key(v1beta1constants.GardenNamespace, v1beta1constants.DaemonsetNameFluentBit), fluentBitService); err != nil {
+				return retry.MinorError(fmt.Errorf("%s service is not ready: %v", v1beta1constants.DaemonsetNameFluentBit, err))
+			}
+			return retry.Ok()
+		}); err != nil {
+			return err
+		}
+
+		// Patching fluent-bit service with the network policy annotations
+		fluentBitServicePatch := client.MergeFrom(fluentBitService.DeepCopy())
+		fluentBitService.ObjectMeta.Annotations = map[string]string{
+			v1beta1constants.AnnotationNetworkPolicyFromPolicyPodLabelSelector: v1beta1constants.NetworkPolicyAllScrapeTargets,
+			v1beta1constants.AnnotationNetworkPolicyFromPolicyAllowedPorts:     "[{\"port\":\"2020\",\"protocol\":\"TCP\"},{\"port\":\"2021\",\"protocol\":\"TCP\"}]",
+		}
+		log.Info("Pathing network policy annotation of the FluentBit Service")
+		if err := seedClient.Patch(ctx, fluentBitService, fluentBitServicePatch); err != nil {
+			return err
+		}
```

Comment:
Won't this patch approach fight with the fluent-operator? I guess when fluent-operator reconciles the fluent-bit resource it will remove the Service annotations, then our seed controller adds them again. How reliable is this?

Reply:
According to the @rfranzke an operator should not remove labels/annotations which are added from outside. I tested it yesterday and the annotations are not removed after the reconciliation of the operator